cmd/run.go

```diff
@@ -0,0 +1,289 @@
+package cmd
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"path/filepath"
+	"runtime"
+	"time"
+
+	"github.com/DataDog/temporalite"
+	"github.com/google/uuid"
+	"github.com/pmezard/go-difflib/difflib"
+	"github.com/urfave/cli/v2"
+	"go.temporal.io/sdk-features/harness/go/cmd"
+	"go.temporal.io/sdk-features/harness/go/history"
+	"go.temporal.io/sdk/client"
+	"go.temporal.io/server/common/log"
+	"go.temporal.io/server/common/log/tag"
+)
+
+func runCmd() *cli.Command {
+	var config RunConfig
+	return &cli.Command{
+		Name:  "run",
+		Usage: "run a test or set of tests",
+		Flags: config.flags(),
+		Action: func(ctx *cli.Context) error {
+			return NewRunner(config).Run(ctx.Context, ctx.Args().Slice())
+		},
+	}
+}
+
+// RunConfig is configuration for NewRunner.
+type RunConfig struct {
+	Lang                string
+	Version             string
+	GenerateHistory     bool
+	DisableHistoryCheck bool
+	// If not set, defaults to creating one
+	Server string
+	// Defaults to unique value
+	Namespace     string
+	RetainTempDir bool
+}
+
+func (r *RunConfig) flags() []cli.Flag {
+	return []cli.Flag{
+		&cli.StringFlag{
+			Name:        "lang",
+			Usage:       "SDK language to run ('go' or 'java' or 'ts')",
+			Required:    true,
+			Destination: &r.Lang,
+		},
+		&cli.StringFlag{
+			Name:        "version",
+			Usage:       "SDK language version to run",
+			Destination: &r.Version,
+		},
+		&cli.BoolFlag{
+			Name:        "generate-history",
+			Usage:       "Generate the history of the features that are run",
+			Destination: &r.GenerateHistory,
```

Comment:
This should be more explicit about the fact that it can overwrite. At least in the usage if not the name

Reply:
:+1: Will do

Reply:
Added.